lib/ansible/constants.py

```diff
@@ -128,12 +137,12 @@ def _get_config(p, section, key, env_var, default):
     if p is not None:
         try:
             value = p.get(section, key, raw=True)
-        except:
+        except configparser.Error:
             pass
 
     if env_var is not None:
         env_value = os.environ.get(env_var, None)
-        if env_value is not None:
+        if env_value is not None and env_value != default:
```

Comment:
Added a special case for this.

It strikes me as odd that ANSIBLE_SQUASH_ACTIONS default is a string, while it's type is a list however.